pkg/docker/config/config.go

```diff
@@ -73,6 +73,59 @@ func SetAuthentication(sys *types.SystemContext, registry, username, password st
 	})
 }
 
+// GetAllCredentials returns the registry credentials for all registries stored
+// in either the auth.json file or the docker/config.json.
+func GetAllCredentials(sys *types.SystemContext) (map[string]types.DockerAuthConfig, error) {
```

Comment:
(Absolutely non-blocking, a weird thought: Would it be more useful to return a `[]byte authConfigFile`, not guaranteeing _anything_ about its format, just that it can be fed back into (…some version of?) this package via `SystemContext.AuthFilePath`?

That would discourage callers from searching in the path, and it would make it possible to add new kinds of credentials in the future or new feeds to `DockerAuthConfig` without the caller having to know — OTOH it would also require the Podman API to pass the `[]byte` as a single unit, and to make the format effectively undocumented in the Podman API, which I suppose is unacceptable.)

Reply:
I concur with the `OTOH` thoughts. We might get into bigger troubles. But certainly I share the desire to not expose such low-level APIs.